apps/user-profile/src/app/pages/settings-page/settings-authorization/mfa-settings/mfa-settings.component.html

```diff
@@ -1,25 +1,48 @@
-<mat-spinner class="ms-auto me-auto mt-2" *ngIf="loadingMfa"></mat-spinner>
-<div [hidden]="loadingMfa">
+<mat-spinner class="ms-auto me-auto mt-2" *ngIf="loadingMfa || loadingCategories"></mat-spinner>
+<div [hidden]="loadingMfa || loadingCategories ">
   <span
     class="toggle-and-button"
     [matTooltip]="errorTooltip | customTranslate | translate"
     [matTooltipDisabled]="mfaAvailable"
     matTooltipPosition="right">
-    <mat-slide-toggle
-      class="toggle-ellipsis"
-      [disabled]="!mfaAvailable"
-      (toggleChange)="toggleEnableMfa()"
-      #toggle
-      >{{'AUTHENTICATION.MFA_TOGGLE' | customTranslate | translate}}
-    </mat-slide-toggle>
-    <button
-      class="show-button"
-      mat-icon-button
-      (click)="showDetailSettings()"
-      [hidden]="!enableDetailSettings"
-      [disabled]="!mfaAvailable">
-      <mat-icon>{{showDetail ? 'keyboard_arrow_up' : 'keyboard_arrow_down'}}</mat-icon>
-    </button>
+    <mat-accordion>
+      <mat-expansion-panel [disabled]="!mfaAvailable || !enableDetailSettings">
+        <mat-expansion-panel-header>
+          <mat-checkbox
+            class="toggle-ellipsis mt-2 me-5"
+            [disabled]="!mfaAvailable"
+            (click)="$event.stopPropagation()"
+            [(ngModel)]="enforceMfa"
+            (change)="toggleEnableMfa()"
+            [indeterminate]="enforceMfa && (!allRpsSelected || categorySelection.selected.length !== categoriesList.length)"
+            >{{'AUTHENTICATION.MFA_TOGGLE' | customTranslate | translate}}
+          </mat-checkbox>
+        </mat-expansion-panel-header>
+        <mat-spinner class="ms-auto me-auto mt-2" *ngIf="loadingCategories"></mat-spinner>
+        <mat-accordion multi *ngIf="!loadingCategories">
+          <mat-expansion-panel *ngFor="let category of categories | keyvalue">
+            <mat-expansion-panel-header>
+              <mat-checkbox
+                class="toggle-ellipsis mt-2"
```

Comment:
I would add also `me-2` class to avoid bad UX near the arrow due to the long category name.
![image](https://user-images.githubusercontent.com/70371638/228554446-b5215961-33af-4ebe-871c-c33da9d9d47e.png)
```suggestion
                class="toggle-ellipsis mt-2 me-2"
```